elections.js

```diff
@@ -6,41 +6,68 @@ const virtualChainId = 1100000;
+        const queryProcessingStarted = await client.createQuery(orbsVotingContractName, "hasProcessingStarted", []);
+        const responseProcessingStarted = await client.sendQuery(queryProcessingStarted);
 
-    const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
-    const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
+        const _isProcessingPeriod = await client.createQuery(orbsVotingContractName, "isProcessingPeriod", []);
+        const responseIsProcessingPeriod = await client.sendQuery(_isProcessingPeriod);
 
-    const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
-    const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
+        const getCurrentEthereumBlockNumberQuery = await client.createQuery(orbsVotingContractName, "getCurrentEthereumBlockNumber", []);
+        const responsegetCurrentEthereumBlockNumber = await client.sendQuery(getCurrentEthereumBlockNumberQuery);
 
-    const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
-    const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
-    const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
+        const ethereumBlockNumberPerOrbs = Number(responsegetCurrentEthereumBlockNumber.outputArguments[0].value);
+        const isProcessingPeriod = Number(responseIsProcessingPeriod.outputArguments[0].value);
+        const inProgress = !(Number(responseProcessingStarted.outputArguments[0].value) === 0);
 
-    const ethResp = await fetch('http://eth4.orbs.com:8080');
-    const eth = await ethResp.json();
+        const ethResp = await fetch('http://eth4.orbs.com:8080');
+        const eth = await ethResp.json();
 
-    return {
-        eth,
-        elections: {
-            ethereumBlockNumberPerOrbs,
-            isProcessingPeriod,
-            next: Number(responseElectionBlock.outputArguments[0].value),
-            inProgress,
-        }
-    };
+        return {
+            ok: true,
+            eth,
+            elections: {
+                ethereumBlockNumberPerOrbs,
+                isProcessingPeriod,
+                next: Number(responseElectionBlock.outputArguments[0].value),
+                inProgress,
+            }
+        };
+    } catch (err) {
+        console.error(`There has been a problem getting the latest elections status due to an error: ${JSON.stringify(err)}`);
+        return {
+            ok: false,
+            err,
+        };
+    }
 }
 
-(async () => {
-    console.log(await getElectionsStatus());
-})();
+const poller = async () => {
+    let result = await getServerElections();
+    if (result.ok) {
+        currentElectionsData = Object.assign({}, result);
+    }
+};
+
+let pid = setInterval(poller, 10 * 1000);
```

Comment:
Are you sure poller cannot be invoked before the previous one finishes like this? 

I would use `setTimeout` and schedule the next one only in the final line of `poller` (and not forget to add a `try{}catch(){}`  to make sure we never get stuck

Reply:
It definitely can happen unless we set an explicit short timeout for the http agents in both the orbs sdk & node-fetch

Due to the fact I cannot set a short timeout for them (at least not for both)
I see your point and I agree it's a safer solution to use setTimeout and manage it on our own. I will come back with such a managed proposition.